### PR TITLE
Env var webhook default for RabbitMQ image

### PR DESCRIPTION
**Pull request by abays**
Since we do not own the RabbitMQ Cluster Operator, we opt to provide a webhook default for its container image via OpenStack Operator (because we cannot modify the RabbitMQ code to use our service operator defaulting pattern).  This removes the need to hardcode the RabbitMQ image in the OpenStack Operator reconciliation code.

### CODE REVIEW

File: apis/core/v1beta1/openstackcontrolplane_types.go [hunk 1]
```diff
@@ -40,6 +41,13 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	// Container image fall-back defaults
+
+	// RabbitMqContainerImage is the fall-back container image for RabbitMQ
+	RabbitMqContainerImage = "registry.redhat.io/rhosp-rhel9/openstack-rabbitmq:17.0"
```
Comment (stuggi) [inline apis/core/v1beta1/openstackcontrolplane_types.go:48]: I think we should change this to `quay.io/podified-antelope-centos9/openstack-rabbitmq:current-podified`

File: config/default/manager_default_images.yaml [hunk 2]
```diff
@@ -13,3 +13,5 @@ spec:
         env:
         - name: OPENSTACKCLIENT_IMAGE_URL_DEFAULT
           value: quay.io/podified-antelope-centos9/openstack-openstackclient:current-podified
+        - name: RABBITMQ_IMAGE_URL_DEFAULT
+          value: registry.redhat.io/rhosp-rhel9/openstack-rabbitmq:17.0
```
Comment (stuggi) [inline config/default/manager_default_images.yaml:17]: I think we should change this to `quay.io/podified-antelope-centos9/openstack-rabbitmq:current-podified`

File: apis/core/v1beta1/openstackcontrolplane_types.go (hunk 1 above)
Comment (abays) [reply apis/core/v1beta1/openstackcontrolplane_types.go:48]: Ack, will do

**Comment by abays** [comment]
> @abays: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> Test name 	Commit 	Details 	Required 	Rerun command
> ci/prow/openstack-operator-build-deploy 	[d973ade](https://github.com/openstack-k8s-operators/openstack-operator/commit/d973ade0657f0e5cab0fc03e56b488a12591e4ee) 	[link](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_openstack-operator/356/pull-ci-openstack-k8s-operators-openstack-operator-main-openstack-operator-build-deploy/1671925360192655360) 	true 	`/test openstack-operator-build-deploy`
> 
> [Full PR test history](https://prow.ci.openshift.org/pr-history?org=openstack-k8s-operators&repo=openstack-operator&pr=356). [Your PR dashboard](https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:abays).

```
{  failed to wait for the created cluster claim to become ready: timed out waiting for the condition}
```

/test openstack-operator-build-deploy

**Pull request merged**